modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/DatafeedRemoteClusterClientIT.java

```diff
@@ -0,0 +1,196 @@
+                        "function": "sum",
+                        "field_name": "bytes"
+                      }
+                    ]
+                  },
+                  "data_description": {
+                    "time_field": "timestamp",
+                    "time_format": "epoch_ms"
+                  },
+                  "analysis_limits": {
+                    "model_memory_limit": "11MB"
+                  },
+                  "model_plot_config": {
+                    "enabled": true,
+                    "annotations_enabled": true
+                  },
+                  "results_index_name": "test_datafeed_out",
+                  "datafeed_config": {
+                    "indices": [
+                      "remote_cluster:remote_index"
+                    ],
+                    "query": {
+                      "bool": {
+                        "must": [
+                          {
+                            "match_all": {}
+                          }
+                        ]
+                      }
+                    },
+                    "runtime_mappings": {
+                      "hour_of_day": {
+                        "type": "long",
+                        "script": {
+                          "source": "emit(doc['timestamp'].value.getHour());"
+                        }
+                      }
+                    },
+                    "datafeed_id": "test_datafeed"
+                  }
+                }""");
+            Response response = localClient.performRequest(request);
+            logger.info("Anomaly Detection Response:", response.getStatusLine());
+
+            request = new Request("GET", "_ml/anomaly_detectors/test_anomaly_detector");
+            response = localClient.performRequest(request);
+            logger.info("Anomaly detection get:", response.getEntity());
+
+            request = new Request("POST", "_ml/anomaly_detectors/test_anomaly_detector/_open");
+            response = localClient.performRequest(request);
+
+            final Request startRequest = new Request("POST", "_ml/datafeeds/test_datafeed/_start");
+            request.setJsonEntity("""
+                {
+                    "start": "2019-04-07T18:22:16Z"
+                }
+                """);
+            try {
+                localClient.performRequest(startRequest);
+                assertTrue("This request should fail", false);
+            } catch (ResponseException e) {
+                String[] messages = e.getMessage().split("\n");
+                assertTrue(Pattern.matches("""
+                    .*Datafeed \\[test_datafeed\\] is configured with a remote index pattern\\(s\\) \
+                    \\[remote_cluster:remote_index\\] but the current node \\[local_cluster-0\\] \
+                    is not allowed to connect to remote clusters. Please enable node.remote_cluster_client \
+                    for all machine learning nodes and master-eligible nodes.*""", messages[1]));
```

Comment:
I wonder if `org.elasticsearch.common.regex.Regex#simpleMatch(java.lang.String, java.lang.String, boolean)` might be more forgiving/readable? it just uses wildcard characters rather than regex syntax.

Reply:
Usually `assertThat(X, containsString(Y))` is good enough 

```assertThat(e.getMessage(), containsString("Datafeed [test_datafeed] is configured with a remote index pattern  [remote_cluster:remote_index] but the current node [local_cluster-0] is not allowed to connect to remote clusters. Please enable node.remote_cluster_client for all machine learning nodes and master-eligible nodes."));```